scalafix-reflect/src/main/scala-3/scalafix/internal/reflect/RuleCompiler.scala

```diff
@@ -0,0 +1,81 @@
+package scalafix.internal.reflect
+
+import dotty.tools.dotc.Compiler
+import dotty.tools.dotc.Run
+import dotty.tools.dotc.core.Contexts.FreshContext
+import dotty.tools.dotc.util.SourceFile
+import dotty.tools.dotc.reporting.StoreReporter
+import dotty.tools.dotc.interactive.InteractiveDriver
+import dotty.tools.io.AbstractFile as DottyAbstractFile
+
+import scala.reflect.io.VirtualDirectory
+import scala.reflect.io.AbstractFile
+import scala.reflect.internal.util.AbstractFileClassLoader
+
+import metaconfig.Configured
+import metaconfig.Input
+import metaconfig.ConfError
+import metaconfig.Position
+
+class RuleCompiler(
+    classpath: String,
+    target: AbstractFile = new VirtualDirectory("(memory)", None)
+) {
+  // we are not configuring any settings!
+  // private val settings = new Settings()
+  // settings.deprecation.value = true // enable detailed deprecation warnings
+  // settings.unchecked.value = true // enable detailed unchecked warnings
+  // settings.outputDirs.setSingleOutput(target)
+  // settings.classpath.value = classpath
+  private val settings = "-unchecked" :: "-deprecation" :: Nil
+  private val driver = new InteractiveDriver(settings.distinct)
+  private val reporter: StoreReporter = new StoreReporter()
+  private given ctx: FreshContext = driver.currentCtx.fresh
+
+  ctx
+    .setReporter(reporter)
+    .setSetting(ctx.settings.outputDir, DottyAbstractFile.getFile(target.path))
+    .setSetting(ctx.settings.classpath, classpath)
+
+  private val compiler: Compiler = new Compiler()
+  private val classLoader: AbstractFileClassLoader =
+    new AbstractFileClassLoader(target, this.getClass.getClassLoader)
+
+  def compile(input: Input): Configured[ClassLoader] = {
+    reporter.removeBufferedMessages(using ctx)
+    val run: Run = compiler.newRun(using ctx)
+    run.compileSources(
+      List(
+        new SourceFile(
+          DottyAbstractFile.getFile(input.path),
+          input.chars
+        )
+      )
+    )
+
+    val errors = reporter.allErrors.map(error => 
+      ConfError
+        .message(error.getMessage)
```

Comment:
is that a multiline error? I guess we'll need unit tests to make sure this is not spamming the error logs for nothing